module/VuFind/tests/unit-tests/src/VuFindTest/Marc/MarcLintTest.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * MarcLint Test Class
+ *
+ * PHP version 7
+ *
+ * Copyright (C) The National Library of Finland 2020-2021.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+namespace VuFindTest\Marc;
+
+/**
+ * MarcLint Test Class
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+class MarcLintTest extends \PHPUnit\Framework\TestCase
```

Comment:
Not sure if you already looked at them, but there are a bunch of test classes (marc_lint_001.phpt through marc_lint_005.phpt) in the [original project](https://github.com/pear/File_MARC/tree/master/tests). Might it be worth porting them over? (I didn't closely review past or current tests, so maybe you've already done this analysis).

Reply:
I did actually overlook them. I've now ported most of the tests over and expanded on that to bring the coverage to 100%. There are minor differences due to how things are handled in MarcReader, but the ported tests are now pretty nice as they're more granular than what I had added, and I also made the result checks more specific than the original ones.